src/marqo/s2_inference/processing/image.py

```diff
@@ -40,6 +41,7 @@
 
 logger = get_logger(__name__)
 
+load_model_lock = threading.Lock()
```

Comment:
Just to double check, are we sure this is the only place chunking is called, that we need to lock?

Reply:
It's the only place I could identify. It does make sense since it's the base class for patching images

Reply:
Got it! @wanliAlex Do you have any insight on this, just to make sure we're locking all the necessary parts of the code?

Reply:
This looks good to me. As you are trying to lock the model loading for `image_preprocessing` models